cmd/drone-docker-ecr/main.go

```diff
@@ -1,5 +1,119 @@
 package main
 
+import (
+	"encoding/base64"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecr"
+)
+
+const defaultRegion = "us-east-1"
+
 func main() {
+	var (
+		repo   = getenv("PLUGIN_REPO")
+		region = getenv("PLUGIN_REGION", "ECR_REGION", "AWS_REGION")
+		key    = getenv("PLUGIN_ACCESS_KEY", "ECR_ACCESS_KEY", "AWS_ACCESS_KEY_ID")
+		secret = getenv("PLUGIN_SECRET_KEY", "ECR_SECRET_KEY", "AWS_SECRET_ACCESS_KEY")
+		create = parseBoolOrDefault(false, getenv("PLUGIN_CREATE_REPOSITORY", "ECR_CREATE_REPOSITORY"))
+	)
+
+	// set the region
+	if region == "" {
+		region = defaultRegion
+	}
+
+	os.Setenv("AWS_ACCESS_KEY_ID", key)
+	os.Setenv("AWS_SECRET_ACCESS_KEY", secret)
+	os.Setenv("AWS_REGION", region)
+
+	sess := session.Must(session.NewSession())
+	svc := ecr.New(sess)
+	username, password, registry, err := getAuthInfo(svc)
+	if err != nil {
+		os.Exit(1)
+	}
+
+	// always attempt to create the repo if createBool is true, eat exists error
```

Comment:
I like that we're documenting that we're eating the exists error, but that's more of an implementation detail for `createRepo`. I was initially confused to see us exiting on err on line 46, since this comment seems to contradict that.

I'd suggest either removing `eat exists error` here, or doing the same and adding something on the `createRepo` call in 44 that is a little more clear:

> "already exists" errors are not returned here, since we want to ensure the repo's existence

Another option would be to rename `createRepo` to `ensureRepoExists` or `ensureRepo` to make this more immediately obvious.

Reply:
This flagged in my head when I was looking it over before submission as well as possibly confusing so I totally understand. I'll rename the function and drop the comment.